controllers/account.lua

```diff
@@ -0,0 +1,28 @@
+local app_helpers = require("lapis.application")
+local validate = require("lapis.validate")
+local capture_errors, assert_error = app_helpers.capture_errors, app_helpers.assert_error
+local Users = require("models.users")
+
+return {
+  before = capture_errors(function(self)
+    self.page_title = self.i18n("account_nav")
+  end),
+  GET = capture_errors(function(self)
+    assert_error(self.session.name, self.i18n("err_not_logged_in"))
+    return { render = "account" }
+  end),
+  POST = capture_errors(function(self)
+    validate.assert_valid(self.params, {
+      { "new_password", exists = true, min_length = 8, i18n("missing_new_password") },
+      { "confirm_new_password", equals = self.params.new_password,
+            i18n("confirmation_password_must_match") },
+    })
+
+    self.params.password = self.params.old_password
+    self.params.name = self.session.name
+    assert_error(Users:verify_user(self.params), i18n("old_password_mismatch"))
```

Comment:
I'm not a big fan of hacking `self.params` to please `Users:verify_user` - the API would be better if the latter just took in the username and password as it expects. Right now, we have to leak `self.params` into it. What do you think @demonnic?

Reply:
That's fine. When I wrote it self.params was just a table containing the username and password. If you want to refactor it to extract the uname and pass from the table and send those specifically go for it.

Reply:
would you rather do that in this PR or another?

Reply:
Another